controllers/upgrade_handlers.go

```diff
@@ -54,9 +59,86 @@ func (b BackupValidationError) Error() string {
 
 // handleUpgrade orchestrate main upgrade steps and update status as needed
 func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
-
 	r.Log.Info("Starting handleUpgrade")
 
+	currentStaterootName, _ := r.RPMOstreeClient.GetCurrentStaterootName()
+	r.Log.Info("stateroots", "current stateroot:", currentStaterootName, "desired stateroot", getDesiredStaterootName(ibu))
+
+	// Checking the stateroot to figure if pre or post reboot. WARNING: the pod may not know if we are boot loop (for now)
+	if getDesiredStaterootName(ibu) != currentStaterootName {
```

Comment:
we can just pass in the `isAfterPivot` down.
It will be easier to write a unittest if we inject this dependency

Reply:
Yeah I was hoping avoid changing the signature...but updated it make easier to test. Created a variable func `isPrePivot` which we can override within a UT. PTAL! e.g 
```
isPrePivot = func(r *ImageBasedUpgradeReconciler, ibu *lcav1alpha1.ImageBasedUpgrade) (bool, error) {
				return true, nil
			}
got, err := r.handleUpgrade(tt.args.ctx, tt.args.ibu)
```

Let me know what you guys think

Reply:
Updated to use ctx

Reply:
After the post pivot this way doesnt seem to work 

```
2023-12-04T18:10:20Z	INFO	controllers.ImageBasedUpgrade	Loaded IBU	{"name": {"name":"upgrade","namespace":"openshift-lifecycle-agent"}, "version": "82808", "desired stage": "Upgrade"}
2023-12-04T18:10:20Z	INFO	controllers.ImageBasedUpgrade	Starting handleUpgrade
2023-12-04T18:10:20Z	ERROR	controllers.ImageBasedUpgrade		{"error": "isAfterPivotKey must be type bool"}
github.com/openshift-kni/lifecycle-agent/controllers.(*ImageBasedUpgradeReconciler).handleUpgrade
	/opt/app-root/src/controllers/upgrade_handlers.go:69
github.com/openshift-kni/lifecycle-agent/controllers.(*ImageBasedUpgradeReconciler).handleStage
	/opt/app-root/src/controllers/ibu_controller.go:199
github.com/openshift-kni/lifecycle-agent/controllers.(*ImageBasedUpgradeReconciler).Reconcile
	/opt/app-root/src/controllers/ibu_controller.go:154
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/opt/app-root/src/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:119
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/opt/app-root/src/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:316
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/opt/app-root/src/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/opt/app-root/src/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227
```

Reply:
I am down with merge it however you prefer it and refactor it later if needed.